fiftyone/utils/eval/coco.py

```diff
@@ -231,3 +473,113 @@ def _iscrowd(detection):
+
+    Args:
+        matches: a list of ``(gt_label, pred_label, iou, pred_confidence)``
+            matches. Either label can be ``None`` to indicate an unmatched
+            object
+        precision: a Numpy array of shape ``(IoU thresholds,
+            classes, len(recall))`` containing precision curves over a range of
+            recall scores for every class and IoU threshold 
+        recall: a 1D array containing recall values related to the given
+            precision
+        classes (None): the list of possible classes. If not provided, the
+            observed ground truth/predicted labels are used
+        missing ("none"): a missing label string. Any unmatched objects are
+            given this label for evaluation purposes
+    """
+
+    def __init__(
+        self, precision, recall, matches, classes=None, missing="none",
+    ):
+        super().__init__(matches, classes=classes, missing=missing)
+        self.precision = precision
+        self.recall = recall
+        self.mAP, self.classwise_AP = self._compute_mAP(self.precision)
+
+    def _compute_mAP(self, precision):
+        classwise_AP = np.mean(precision, axis=(0, 2))
+        existing_classes = classwise_AP > -1
+
+        if len(existing_classes) == 0:
+            mAP = -1
+        else:
+            mAP = np.mean(classwise_AP[existing_classes])
+
+        return mAP, classwise_AP
+
+    def plot_pr_curves(self, classes=None, ax=None, block=False, **kwargs):
+        """Plot precision-recall (PR) curves for COCO detection results.
+
+        Args:
+            classes (None): the classes to generate plots for, by default will
+                plot the top 3 AP class
+            average ("micro"): the averaging strategy to use when computing
+                average precision
+            ax (None): an optional matplotlib axis to plot in
+            block (False): whether to block execution when the plot is
+                displayed via ``matplotlib.pyplot.show(block=block)``
+            **kwargs: optional keyword arguments for
+                ``sklearn.metrics.PrecisionRecallDisplay.plot(**kwargs)``
+
+        Returns:
+            a list of the matplotlib axes containing the plots
+        """
+        if not classes:
+            print(
+                "No classes specified, plotting PR curve of highest AP class"
+            )
+            max_class_inds = np.argsort(self.classwise_AP)[::-1][:3]
+            classes = list(np.array(self.classes)[max_class_inds])
+
+        displays = []
+        for c in classes:
+            class_ind = self.classes.index(c)
+            precision = np.mean(self.precision[:, class_ind], axis=0)
+            recall = self.recall
+            avg_precision = np.mean(precision)
+            display = skm.PrecisionRecallDisplay(
+                precision=precision, recall=recall
+            )
+            label = "AP = %.2f, class = %s" % (avg_precision, c)
+            display.plot(ax=ax, label=label, **kwargs)
```

Comment:
Shouldn't there be a single `ax` return value that is the axes in which all PR curves are plotted?